### ISSUE TITLE
`Trainer.fit` stopped: `max_epochs=1` reached.

### ISSUE THREAD

**Issue by yubin-ai** [scverse/scvi-tools#2420]
[`Trainer.fit` stopped: `max_epochs=1` reached.]

```python
scvi.model.SCVI.setup_anndata(
ad,
layer="raw",
categorical_covariate_keys=['block_id','tma'])
model = scvi.model.SCVI(ad, n_layers=2, n_latent=30)

model.train(enable_progress_bar=False)
SCVI_LATENT_KEY = "X_scVI"
```

GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
You are using a CUDA device ('NVIDIA A10G') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
`Trainer.fit` stopped: `max_epochs=1` reached.

#### Versions:
'1.0.2'

> VERSION

in scvi.train.Trainer,   max_epochs is the  Number of passes through the dataset. If None, defaults to np.min([round((20000 / n_cells) * 400), 400]). so when the dataset is larger than 20k*400 it will be 1. I have a very large dataset and I assume it leads to max_epochs =1 .

is this ok or this is wrong?

**Comment by canergen** [comment]
I recommend setting it manually to 20. With 20 epochs we have good experience on large datasets. You might want to increase batch size to 1024 to increase train speed and enable early stopping, see documentation on how to enable it.
Changing the default creates backwards incompatibility and I am not eager to change it for now.

**Comment by yubin-ai** [comment]
> I recommend setting it manually to 20. With 20 epochs we have good experience on large datasets. You might want to increase batch size to 1024 to increase train speed and enable early stopping, see documentation on how to enable it. Changing the default creates backwards incompatibility and I am not eager to change it for now.

This is super helpful! Thanks.
I understand the concern on changing the default. How about adding a detection step on the cell number and then add a warning there to inform the user this potential issue. I had a hard time understanding the results due to its super weird looking umap and did not realize this issue until several rounds of debugging. and the dataset is so big, it can be hard to debug. Appreciate it!

**Comment by martinkim0** [comment]
Thanks for the suggestion! We're are aware of this potential issue - it's been addressed in the latest version (#2083) that is set to release soon (1.1.0). It will warn when `max_epochs=1` due to a larger number of observations.

**Comment by martinkim0** [comment]
Closing since this is addressed in the new release.
